### DISCUSSION TITLE
Prediction

### DISCUSSION THREAD

**Discussion by johncrater** [dyn4j/dyn4j#300] [Q&A]
I have a question. In my game. one of the bodies needs to be able to "catch" a ball thrown or kicked by another. This involves predicting where the ball is going to be and when. I'm applying an impulse for the throwing/kicking of the ball. But, I'm not sure where exactly it is going to end up so I can direct one of the bodies to the correct destination. Is there such a mechanism in the game to predict where a body will be, baring collisions naturally, after a given number of steps. I could probably do it myself since there are only a few variables, linear velocity, linear dampening, I also account for air resistance and of course gravity. But, that would have me duplicating a lot of what the physics engine is already doing. Any ideas?

**Comment by wnbittle** [comment]
In the [BasketBall example - watch towards the end a trajectory is toggled on](https://github.com/dyn4j/dyn4j-samples/blob/747a479d3a0059f73a4de62d33b6003d387d933e/src/main/java/org/dyn4j/samples/BasketBall.java#L507), I do something similar where I'm predicting the ball's center point based on the settings before launch:

https://github.com/dyn4j/dyn4j-samples/blob/747a479d3a0059f73a4de62d33b6003d387d933e/src/main/java/org/dyn4j/samples/BasketBall.java#L507

It may not be the same as what you need, but it should give you some ideas.  It's definitely a calculation you'd have to do on your side since the engine itself isn't doing any prediction.  One element it does incorporate is the iterative nature of the engine - for example, in the BasketBall example if you simply use ballistic motion equations, you'll get a different answer than the engine.
